packages/modern-test-app/app/components/helpers.ts

```diff
@@ -0,0 +1,11 @@
+import Component from '@glimmer/component';
+
+export default class extends Component {
+  get objectOrUndefined(): object | undefined {
+    return new Date().getMonth() === 7 ? { foo: 'bar' } : undefined;
+  }
+
+  get arrayOrUndefined(): object | undefined {
```

Comment:
this isn't an array :sweat_smile:

Reply:
yeap! it is. I had a huge doubts about this approach in the first place and more I think about it, the more I get confident this is totally wrong.
I think #196 if the way to go (just need to make CI happy)